nni/algorithms/compression/pytorch/pruning/admm_pruner.py

```diff
@@ -159,7 +161,7 @@ def callback():
             # step 1: optimize W with AdamOptimizer
             for epoch in range(self._training_epochs):
                 self._trainer(self.bound_model, optimizer=optimizer,
-                              criterion=criterion, epoch=epoch, callback=callback)
+                              criterion=self._criterion, epoch=epoch, callback=callback)
```

Comment:
can we handle the criterion in trainer?

Reply:
Criterion should be defined by the user, so it has to be passed as a parameter.